src/cascadia/Remoting/WindowManager.cpp

```diff
@@ -103,24 +107,58 @@ namespace winrt::Microsoft::Terminal::Remoting::implementation
                                   "WindowManager_ProposeCommandline",
                                   TraceLoggingBoolean(_shouldCreateWindow, "CreateWindow", "true iff we should create a new window"),
                                   TraceLoggingPointer(nullptr, "Id", "No ID provided"),
+                                  TraceLoggingWideString(givenName.c_str(), "Name", "The name we should assign this window"),
                                   TraceLoggingLevel(WINEVENT_LEVEL_VERBOSE));
             }
         }
         else
         {
             // We're the monarch, we don't need to propose anything. We're just
             // going to do it.
-            TraceLoggingWrite(g_hRemotingProvider,
-                              "WindowManager_ProposeCommandline_AsMonarch",
-                              TraceLoggingBoolean(_shouldCreateWindow, "CreateWindow", "true iff we should create a new window"),
-                              TraceLoggingLevel(WINEVENT_LEVEL_VERBOSE));
+            //
+            // However, we _do_ need to ask what our name should be. It's
+            // possible someone started the _first_ wt with something like `wt
+            // -w king` as the commandline - we want to make sure we set our
+            // name to "king".
+            auto findWindowArgs{ winrt::make_self<Remoting::implementation::FindTargetWindowArgs>(args) };
+            _raiseFindTargetWindowRequested(nullptr, *findWindowArgs);

```

Comment:
Worried about `ProposeCommandline` emitting `FindTargetWindowRequested` events. We sure that's okay? I understand the comment, but I do not know _why_ doing this accomplishes the comment's thesis.

Reply:
Yea, the `FindTargetWindow` event is the `WindowManager`'s way of saying "I do not know how to figure out how to turn this list of args into a window ID/name. Whoever's listening to this event _does_, so I'll ask them". It's a convoluted way of hooking the `WindowManager` up to `AppLogic` without actually telling it anything about `TerminalApp` (or even `WindowsTerminal`)

Reply:
(I'm adding that as a comment for clarity)